File: BookNestFrontend/booknest_desktop/lib/models/reading_status.dart
class ReadingStatus {
  final int id;
  final String name;

  ReadingStatus({required this.id, required this.name});

  factory ReadingStatus.fromJson(Map<String, dynamic> json) {
    return ReadingStatus(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
    );
  }
}

File: BookNestBackend/BookNest.Services/Database/Entities/ReadingStatus.cs
using System.ComponentModel.DataAnnotations;

namespace BookNest.Services.Database.Entities
{
    public class ReadingStatus
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string Name { get; set; } = string.Empty;

        public ICollection<TBRList> TBRLists { get; set; } = new List<TBRList>();
    }
}
